tools/pickles_to_json.py

```diff
@@ -0,0 +1,399 @@
+    """
+    Load and combine all partitions of a pickled item from DynamoDB.
+    :param source_table: The DynamoDB table resource.
+    :param key: The primary key of the item to retrieve.
+    :return: The combined pickled data as bytes.
+    """
+    response = source_table.get_item(Key={"key": key, "index": 0}, ConsistentRead=True)
+    if "Item" not in response:
+        raise Exception(f"No item found for key {key} at index 0")
+    item = response["Item"]
+    num_partitions = int(item.get("num_partitions", 1))
+    combined_data = bytearray()
+    for index in range(num_partitions):
+        response = source_table.get_item(Key={"key": key, "index": index}, ConsistentRead=True)
+        if "Item" not in response:
+            raise Exception(f"Missing partition {index} for key {key}")
+        item = response["Item"]
+        combined_data.extend(item["val"].value)
+    return bytes(combined_data)
+
+
+def dump_pickle_key(source_table: ServiceResource, key: str) -> None:
+    """
+    Load the pickled data from DynamoDB for a given key, handling partitioned
+    items, and print the full pickle data.
+    :param source_table: The DynamoDB table resource.
+    :param key: The primary key of the item to retrieve.
+    """
+    try:
+        pickled_data = combine_pickle_partitions(source_table, key)
+        loaded_pickle = pickle.loads(pickled_data)
+        print(f"Key: {key} - Pickle data:")
+        print(loaded_pickle)
+    except Exception as e:
+        print(f"Key: {key} - Failed to load pickle: {e}")
+        raise
+
+
+def dump_pickle_keys(source_table: ServiceResource, keys: List[str]) -> None:
+    """
+    Load and print pickles for the given list of keys.
+    :param source_table: The DynamoDB table resource.
+    :param keys: A list of keys for which to load and print pickles.
+    """
+    for key in keys:
+        dump_pickle_key(source_table, key)
+
+
+def dump_json_key(source_table: ServiceResource, key: str) -> None:
+    """
+    Load the JSON data from DynamoDB for a given key and print it.
+    :param source_table: The DynamoDB table resource.
+    :param key: The primary key of the item to retrieve.
+    """
+    try:
+        json_data = combine_json_partitions(source_table, key)
+        if json_data is not None:
+            print(f"Key: {key} - JSON data:")
+            print(json_data)
+        else:
+            print(f"Key: {key} - No JSON value found")
+    except Exception as e:
+        print(f"Key: {key} - Failed to load JSON: {e}")
+
+
+def dump_json_keys(source_table: ServiceResource, keys: List[str]) -> None:
+    """
+    Load and print JSON data for the given list of keys.
+    :param source_table: The DynamoDB table resource.
+    :param keys: A list of keys for which to load and print JSON data.
+    """
+    for key in keys:
+        dump_json_key(source_table, key)
+
+
+def delete_keys(source_table: ServiceResource, keys: List[str]) -> None:
```

Comment:
Based on discussion with @nemacysts and @EmanElsaban we can probably do something here where we look for jobs that Tron doesn't know about and use that as the test for a valid delete